src/frontend/web_application/src/components/MessageList/components/Message/presenter.jsx

```diff
@@ -62,20 +39,19 @@ class Message extends Component {
     locale: undefined,
   };
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      body: '',
-      isExpanded: true,
-      isTooLong: false,
-      isActive: false,
-    };
-    this.handleActiveClick = this.handleActiveClick.bind(this);
-    this.handleExpandClick = this.handleExpandClick.bind(this);
+  state = {
+    bodyHeight: null,
+    isTooLong: false,
+    isFold: false,
+    isUnread: false,
+  };
+
+  componentWillMount() {
+    this.setState(prevState => generateStateFromProps(this.props, prevState));
   }
 
   componentDidMount() {
-    this.setBodyHeight();
+    this.setContentHeight();
```

Comment:
I didn't saw this but you might have a warning here, if I'm right.

You cannot setState in `componentDidMount` nor `render` methods because it produce a new render cycle. So in some cases it induces to an infinite loop.

For now I'm sure it is not a good practice unless setState is used in a setTimeout.

Reply:
ok if you can't deal with in this PR (but we will have to fix it soon)